llvm/utils/UpdateTestChecks/common.py

```diff
@@ -2358,7 +2358,7 @@ def add(var):
         r"(?<=\")(.+ )?(\w+ version )[\d.]+(?:[^\" ]*)(?: \([^)]+\))?",
         r"{{.*}}\2{{.*}}",
     ),  # preface with glob also, to capture optional CLANG_VENDOR
-    (r'(!DIFile\(filename: ".+", directory: )".+"', r"\1{{.*}}"),
+    (r'(!DIFile\(filename: ")(.+/)?([^/]+", directory: )".+"', r"\1{{.*}}\3{{.*}}"),
```

Comment:
Do we get file paths in the filename field? (off the top of my head I can't remember - I assume that's why you've added the `{{.*}}` prefix to file base names here?)

Reply:
Yes, and in particular update_cc_test_checks.py seems to invoke clang in such a way as to always generate absolute paths.

The approach I took leaves some potential for ambiguity in some cases, but I think real issues would only crop up for largely contrived test cases (like having a test which depends on multiple input files at different relative paths but with the same filename) and even then the order of the metadata in the output should be stable, so it should be pretty inconsequential.